### ISSUE TITLE
PACE_PAX Punpy/comet-maths cannot correlate with 0 std

### ISSUE THREAD

**Issue by oceancolorcoder** [nasa/HyperCP#274]
@ARamsay17 the L1BQC is on the Box:
PACE-PAX_PACE-PAX_20240912_200000_L1BQC.hdf
Setup:
![image](https://github.com/user-attachments/assets/cfa92dab-1255-42b6-9283-18ce420e3384)
Error:
```
25 spectra in slice (ensemble).
2 spectra remaining in slice to average after filtering to lowest 10.0%.
Calculating M99 glint correction with complete LUT
Calculating Zhang glint correction.
Zhang17 Elapsed Time: 3.4 s
Reading : Data/hybrid_reference_spectrum_p1nm_resolution_c2020-09-21_with_unc.nc
Traceback (most recent call last):
  File "/Users/daurin/GitRepos/HyperCP/Main.py", line 515, in singleL2Clicked
    self.processSingle("L2")
  File "/Users/daurin/GitRepos/HyperCP/Main.py", line 484, in processSingle
    Controller.processFilesSingleLevel(
  File "/Users/daurin/GitRepos/HyperCP/Source/Controller.py", line 921, in processFilesSingleLevel
    Controller.processSingleLevel(pathOut, fp, calibrationMap, level)                
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/Controller.py", line 778, in processSingleLevel
    root = Controller.processL2(root,outFilePath)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/Controller.py", line 460, in processL2
    node = ProcessL2.processL2(root,station)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/ProcessL2.py", line 2359, in processL2
    if not ProcessL2.stationsEnsemblesReflectance(node, root,station):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/ProcessL2.py", line 2248, in stationsEnsemblesReflectance
    if not ProcessL2.ensemblesReflectance(node, sasGroup, referenceGroup, ancGroup, 
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/ProcessL2.py", line 1801, in ensemblesReflectance
    xUNC.update(instrument.ClassBasedL2(node, uncGroup, rhoScalar, rhoVec, rhoUNC, waveSubset, xSlice))
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/ProcessInstrumentUncertainties.py", line 553, in ClassBasedL2
    p_unc.pie_plot_class_l2(
  File "/Users/daurin/GitRepos/HyperCP/Source/Uncertainty_Visualiser.py", line 131, in pie_plot_class_l2
    results, values = self._engine.breakdown_Class_L2(rrs_vals, lw_vals, rrs_uncs, lw_uncs, False)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/Uncertainty_Visualiser.py", line 405, in breakdown_Class_L2
    output['Lw'][keys_lw[indx]] = self.punpy_MCP.Propagate_Lw_HYPER(lw_vals, uLw)
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/Uncertainty_Analysis.py", line 164, in Propagate_Lw_HYPER
    return self.MCP.propagate_random(self.Lw,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/hypercp/lib/python3.11/site-packages/punpy/mc/mc_propagation.py", line 123, in propagate_random
    return self.propagate_standard(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/hypercp/lib/python3.11/site-packages/punpy/mc/mc_propagation.py", line 559, in propagate_standard
    MC_x = self.generate_MC_sample(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/hypercp/lib/python3.11/site-packages/punpy/mc/mc_propagation.py", line 630, in generate_MC_sample
    MC_data = cm.correlate_sample_corr(MC_data, corr_between)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/hypercp/lib/python3.11/site-packages/comet_maths/generate_sample/generate_sample.py", line 773, in correlate_sample_corr
    raise ValueError(
ValueError: comet_maths.generate_sample.correlate_sample_corr: you cannot correlate a sample with 0 std
```

**Comment by oceancolorcoder** [comment]
This "sample with 0 std" is an issue with numerous PACE-PAX files.

**Comment by oceancolorcoder** [comment]
Thanks for checking. I will follow up on #267 presently while you chase down the 0 std #274 issue.

**Comment by ARamsay17** [comment]
@oceancolorcoder I also noticed that Krista does not run the uncertainty visualisation. It could be that this has the same cause as #267

**Comment by ARamsay17** [comment]
@oceancolorcoder I commented out the following lines (650-651) from ProcessL2.filterdata:

if badTimes == []:
        startLength = 1 # avoids div by zero below when finalCount is 0

They don't seem to have a purpose and are creating the error. Once removed the PACE_PAX 20h will run. I'll upload the L2 hdf and plots to the box for further checks.

**Comment by oceancolorcoder** [comment]
@ARamsay17 that is not the problem with issue #274. That problem with badTimes was fixed yesterday for issue #267 . The patch is merged to nasa:dev. Please undo your changes to PL2 prior to pull to avoid conflict on those lines.

**Comment by oceancolorcoder** [comment]
This issue is a problem with the visualizer related to "comet-maths cannot correlate with 0 std"

**Comment by ARamsay17** [comment]
@oceancolorcoder I was unable to fully pin down this issue. My solution for now is to catch the error in uncertainty_visualisation and warn the user that the uncertainty breakdown failed in this case. I will merge with NASA:dev now, thank you for the update.

**Comment by oceancolorcoder** [comment]
resolved with #284
